heroic-test-it/src/main/java/com/spotify/heroic/test/AbstractMetricBackendIT.java

```diff
@@ -93,21 +96,61 @@ public void evaluate() throws Throwable {
                 base.evaluate();
                 core.shutdown().get();
             } else {
-                log.info("Omitting "  + description + " since module is not configured");
+                log.info("Omitting " + description + " since module is not configured");
             }
         }
     };
 
     @Test
-    public void testWrite() throws Exception {
+    public void testReadWriteEdges() throws Exception {
         // write and read data back
-        final MetricCollection points = Data.points().p(100000L, 42D).build();
+        final MetricCollection expected =
+            Data.points().p(101L, 42D).p(199L, 42D).p(200L, 42D).build();
+
+        final MetricCollection points = Data
+            .points()
+            .p(99L, 42D)
+            .p(100L, 42D)
+            .p(101L, 42D)
+            .p(199L, 42D)
+            .p(200L, 42D)
+            .p(201L, 42D)
+            .build();
+
         backend.write(new WriteMetric.Request(s1, points)).get();
         FetchData data = backend
-            .fetch(new FetchData.Request(MetricType.POINT, s1, new DateRange(10000L, 200000L),
+            .fetch(new FetchData.Request(MetricType.POINT, s1, new DateRange(100L, 200L),
                 QueryOptions.builder().build()), FetchQuotaWatcher.NO_QUOTA)
             .get();
 
-        assertEquals(ImmutableSet.of(points), ImmutableSet.copyOf(data.getGroups()));
+        assertEquals(ImmutableSet.of(expected), ImmutableSet.copyOf(data.getGroups()));
+    }
+
+    @Ignore("check that this works when intervals are strongly defined")
+    @Test
+    public void testPeriodReadWrite() throws Exception {
+        final Optional<Long> maybePeriod = period();
+        assumeTrue("has period", maybePeriod.isPresent());
+        final long count = 5;
+        final long period = maybePeriod.get();
+
+        final Points points = Data.points();
+
+        for (int i = 1; i < count; i++) {
+            points.p(i * period, 1.0D);
+            points.p((i + 1) * period - 1, 1.0D);
```

Comment:
I believe it could be useful to clarify this test. 
* I'm unsure if the concept of "period" is clear to everyone else - to me it's a bit unclear what it is and why we're choosing to insert test-points at these specific time intervals.